katas/content/KatasLibrary.qs

```diff
@@ -2,9 +2,93 @@
 // Licensed under the MIT License.
 
 namespace Microsoft.Quantum.Katas {
+    open Microsoft.Quantum.Arrays;
     open Microsoft.Quantum.Diagnostics;
     open Microsoft.Quantum.Intrinsic;
 
+    /// # Summary
+    /// Given two operations, checks whether they act identically for all input states.
+    /// This operation is implemented by using the Choi–Jamiołkowski isomorphism.
+    operation CheckOperationsEquivalence(
+        op : (Qubit[] => Unit is Adj + Ctl),
+        reference : (Qubit[] => Unit is Adj + Ctl),
+        inputSize: Int)
+    : Bool {
+        Fact(inputSize > 0, "`inputSize` must be positive");
+        use (control, target) = (Qubit[inputSize], Qubit[inputSize]);
```

Comment:
Why do we need `inputSize` number of control qubits here? That makes the resulting operation much more expensive, and I don't think it gives us any additional validation compared to using a single qubit. Is there some corner case I'm missing?

Reply:
This is the same implementation as [AssertOperationsEqualReferenced](https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.assertoperationsequalreferenced).

This is a generalization of the check for equivalence for n-qubits instead of just one, hence the input size parameter. Since `main` doesn't have many katas yet, we are using this to check operations that act on just 1 qubit, but it is/will be also used to check operations that act on multiple qubits. I think it's ok to have just one general implementation instead of two implementations, one for 1-qubit operations and another one for n-qubit operations.

The check could also be done using the approach of [AssertOperationsEqualInPlace](https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.assertoperationsequalinplace) which does not need twice as many qubits, but it requires multiple calls to the operations being compared.

I believe the current implementation is the appropriate one for katas because most (maybe all) the katas just require a few qubits, so we generally won't be using more than 6 qubits to check solutions. In my opinion, this implementation is also easier to understand since it directly maps to the Choi–Jamiołkowski isomorphism.

Reply:
@swernli based on our offline conversation, I created #535 and referenced in the doc comments of the operation.